Java/core/src/main/java/com/amazon/randomcutforest/executor/ITraversable.java

```diff
@@ -79,4 +81,11 @@
      *         traversal query, otherwise false.
      */
     boolean isOutputReady();
+
+    default double dynamicScore(double[] point, int ignoreMass, BiFunction<Double, Double, Double> scoreSeen,
```

Comment:
If we want to special-case the anomaly scoring path, my suggestion would be to create a new anomaly score visitor (possibly a sub-interface of Visitor, but not necessarily) that defines `scoreSeen` etc. as methods. Then add a method to ITraversable that takes an anomaly score visitor.

I think there's potentially an even more generic solution here, where we define "lightweight" and "heavyweight" visitors, where the lightweight visitor has access to exactly what the anomaly scoring algorithm needs to work, and the heavyweight visitors get to act on the full node view context.

The name `dynamicScore` is not very descriptive in the context of a Traversable.

Reply:
Agree with all three points -- did not edit visitors in this PR (it's already large). At the current moment this provides a mechanism for us to measure overhead of the visitor. The name comes from the DynamicScoringRandomCutForest (since the 1.0) which we should absorb (eventually, and not in this PR). That would enable a simple path to using "displacement score" discussed in the original paper. But yes, the name is up for discussion (both from the forest context as well as from Traversable).

Reply:
yes - done (removed from the interface - but the tree classes still needs to have them while we compare different paths)